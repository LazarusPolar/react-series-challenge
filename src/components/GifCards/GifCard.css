.GifCards {
    z-index: 10px;
    display: inline-block;
}

.GifCards img {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}