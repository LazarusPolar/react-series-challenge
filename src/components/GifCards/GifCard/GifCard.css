.GifCard {
    z-index: 10px;
    display: inline-block;
}

.GifCard img {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}