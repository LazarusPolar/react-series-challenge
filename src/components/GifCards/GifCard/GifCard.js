import React from 'react';

const GifCard = (props) => {

}

export default GifCard;