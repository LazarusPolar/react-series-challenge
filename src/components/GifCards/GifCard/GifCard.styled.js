import styled from 'styled-components';


export const StyledGifCard = styled.div`
    display: inline;
    text-align: center;
`

export const StyledImage = styled.img`
    width: 200px;
    height: 150px;
    display: inline;
`