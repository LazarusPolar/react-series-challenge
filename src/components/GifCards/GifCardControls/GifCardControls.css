.GifCardControls {
    z-index: 10px;
    display: inline-block;
}

.GifCardControls button {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}