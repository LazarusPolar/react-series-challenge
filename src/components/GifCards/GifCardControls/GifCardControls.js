import React from 'react';

const GifCardControls = (props) => {
    //console.log("props", props);

    return(
        <div className = "GifCardControls">
            <button onClick = {props.addRemoveGif}> {props.isFavorite ? <h1>Quitar Favorito</h1> : <h1>Poner Favorito</h1>} </button>
        </div>
    );
}

export default GifCardControls;