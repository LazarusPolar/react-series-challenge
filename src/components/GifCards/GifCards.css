.GifCards {
    z-index: 10px;
    display: inline-block;
}

.GifCards img {
    width: 250px;
    height: 150px;
    z-index: 10px;
    display: inline-block;
}