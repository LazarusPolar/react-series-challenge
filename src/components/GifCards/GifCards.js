import React from 'react';
import GifCard from '../GifCards/GifCard/GifCard'
import GifCardControls from '../GifCards/GifCardControls/GifCardControls'
import './GifCards.css';

const GifCards = (props) => {

        let arrays = null;
        const { gifData, searchedValue, searchedGifs } = props; 
        //console.log("props", props); 

        if(gifData !== null && gifData !== undefined){
            arrays = Object.keys(gifData).map((key, index) => {
                let { id, url } = gifData[index];

                console.log("url: " + url);
                //return (<img key = {id} src = {url} alt = "Cargando..." />);
                return (
                <div key = {id} className = "GifCards">
                <GifCard 
                    //loadError = {props.loadError}
                    //loadSuccess = {props.loadSuccess}
                    imageUrl = {url}
                     />
                    <GifCardControls key = {index} searchedGifs = {() => searchedGifs({ id: id, searchedValue: searchedValue, url: url})} />
                </div>);
            });

            //console.log("Arreglo" , arrays[0]);
            if(arrays.length === 0){
                arrays = <h1> No results :( </h1>;
            }
        }

        /*
        if(this.state.isLoading) {
            image = <p> Cargando imagen... </p>
        } else {
            //TODO: Spinner
            loading = <p> Cargando </p>
            image = <img src = {this.props.imageUrl} onLoad = {this.handleOnLoad} onError = {this.handleError} />
        }*/

        return(
            <div className = "GifCards">
                { arrays }
            </div>
        )
    
}

export default GifCards;