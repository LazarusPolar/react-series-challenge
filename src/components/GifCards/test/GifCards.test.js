import GifCards from '../GifCards';
import GifCard from '../GifCard/GifCard'
import GifCardControls from '../GifCardControls/GifCardControls'
import React, {Fragment} from 'react';
import { GifCardsStyled, StyledH1 } from '../GifCards.styled';
import { shallow, mount, render } from 'enzyme';

describe('GifCards', () => {
  const func = jest.fn();
  const searchResult = [{
      id: 1,
      searchedValue: "trendy",
      url: "https://image", 
      isFavorite: true,
  }]
  const searchResultEmpty = [];
  const themeSunny = {
    gifPageBackground: '#969ec3',
    favoritePageBackground: '#c8bed7',
    headerBackground: '#fde1cc',
    cardBackground: '#fec492',
}
let wrapper;


    beforeEach(() => {
        wrapper = shallow(
            <GifCards
            gifData = { searchResult } 
            addRemoveGif = { func } 
            searchedValue = { searchResult.searchedValue }/>
        );
    });

    test('should set GifCards ', () => {

        const comp = render(
            <Fragment>
                <GifCardsStyled key = { searchResult.id } {...themeSunny}>
                    <GifCard imageUrl = { searchResult.url }/>

                    <GifCardControls 
                    key = {1} 
                    addRemoveGif = { func } 
                    isFavorite = { searchResult.isFavorite } />
                </GifCardsStyled>
            </Fragment>)

        expect(comp).toHaveLength(1);
    });

    test('should not set GifCards ', () => {
        expect(wrapper.contains(
            <Fragment>
                <GifCardsStyled key = { searchResultEmpty.id } {...themeSunny}>
                    <GifCard imageUrl = { searchResultEmpty.url }/>

                    <GifCardControls 
                    key = {1} 
                    addRemoveGif = { func } 
                    isFavorite = { searchResultEmpty.isFavorite } />
                </GifCardsStyled>
            </Fragment>
        )).toBe(false);
    });

})
