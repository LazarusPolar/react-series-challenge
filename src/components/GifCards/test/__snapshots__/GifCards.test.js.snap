// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCards should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GifCards
    addRemoveGif={[Function]}
    gifData={
        Array [
            Object {
              "id": 1,
              "isFavorite": true,
              "searchedValue": "trendy",
              "url": "https://image",
            },
          ]
    }
    searchedValue={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <styled.div
          addRemoveGif={[Function]}
          gifData={
                    Array [
                              Object {
                                "id": 1,
                                "isFavorite": true,
                                "searchedValue": "trendy",
                                "url": "https://image",
                              },
                            ]
          }
          searchedValue={undefined}
>
          <GifCard
                    imageUrl="https://image"
          />
          <GifCardControls
                    addRemoveGif={[Function]}
                    isFavorite={true}
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "addRemoveGif": [Function],
          "children": Array [
            <GifCard
              imageUrl="https://image"
/>,
            <GifCardControls
              addRemoveGif={[Function]}
              isFavorite={true}
/>,
          ],
          "gifData": Array [
            Object {
              "id": 1,
              "isFavorite": true,
              "searchedValue": "trendy",
              "url": "https://image",
            },
          ],
          "searchedValue": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "imageUrl": "https://image",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "addRemoveGif": [Function],
              "isFavorite": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <styled.div
            addRemoveGif={[Function]}
            gifData={
                        Array [
                                    Object {
                                      "id": 1,
                                      "isFavorite": true,
                                      "searchedValue": "trendy",
                                      "url": "https://image",
                                    },
                                  ]
            }
            searchedValue={undefined}
>
            <GifCard
                        imageUrl="https://image"
            />
            <GifCardControls
                        addRemoveGif={[Function]}
                        isFavorite={true}
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "addRemoveGif": [Function],
            "children": Array [
              <GifCard
                imageUrl="https://image"
/>,
              <GifCardControls
                addRemoveGif={[Function]}
                isFavorite={true}
/>,
            ],
            "gifData": Array [
              Object {
                "id": 1,
                "isFavorite": true,
                "searchedValue": "trendy",
                "url": "https://image",
              },
            ],
            "searchedValue": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "imageUrl": "https://image",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "addRemoveGif": [Function],
                "isFavorite": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
